base_location/models/res_partner.py

```diff
@@ -27,13 +27,28 @@ def _onchange_city_id(self):
                     'zip_id': [('city_id', '=', self.city_id.id)]
                 },
             }
+        if self.country_id:
```

Comment:
```suggestion
        if self.country_id and self.contry_enforce_cities:
```

Reply:
This follows https://github.com/OCA/partner-contact/commit/64aaafa45c022116084675e6181d338a8c386f83 and it's for avoiding empty search, as `city_id` is not filled in that cases.

Reply:
We don't risk to get an empty search as we return zip based on the selected country. The data-model guarantee that a `zip.city` object have a related country.

IMO, it make sense that we have empty search if the user select a country that he didn't set zips for.

If for some use cases, we want to show all zips even the user select a country, I believe `enforce_cities` is not a good criteria. A new field `restrict_zip_to_country` make more sense.

Reply:
OK, I didn't remember `city_id` is now required for zips in the new data model. OK then